### PR TITLE
perf(collaboration): add check permission for aad plugin

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/aad/constants.ts [hunk 1]
```diff
@@ -20,6 +20,16 @@ export class Constants {
   static statusCodeUserError = 400;
   static statusCodeServerError = 500;
 
+  static createOwnerDuplicatedMessage =
+    "One or more added object references already exist for the following modified properties: 'owners'.";
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/resource/aad/constants.ts:24]: This error message is very technical, we'd better use another words to general users such "xxx has already be added to the owner list"

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/constants.ts:24]: This error message is from Graph, and is used to match the scenario that we add a user who is already owner. Should be added when adding grant api. I will remove this line.

File: packages/fx-core/src/plugins/resource/aad/graph.ts [hunk 2]
```diff
@@ -111,6 +111,34 @@ export namespace GraphClient {
...
+    }
+
+    if (!userObjectId) {
+      throw new Error(
+        `${GraphClientErrorMessage.CheckPermissionFailed}: ${GraphClientErrorMessage.UserObjectIdIsNull}.`
+      );
+    }
+
+    const instance = initAxiosInstance(graphToken);
+    const response = await instance.get(
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/resource/aad/graph.ts:132]: Do we need to add try catch block in case of network error and other unknown errors?

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/graph.ts:132]: All the errors here will be caught in AadAppClient.ts and throw as user/system error. We don't need try catch here.

File: packages/fx-core/src/plugins/resource/aad/plugin.ts [hunk 3]
```diff
@@ -321,6 +326,34 @@ export class AadAppForTeamsImpl {
...
+
+    const result = [
+      {
+        name: Constants.permissions.name,
+        type: Constants.permissions.type,
+        roles: isAadOwner ? [Constants.permissions.owner] : [Constants.permissions.noPermission],
+        resourceId: config.objectId!,
+      },
+    ];
+    Utils.addLogAndTelemetry(ctx.logProvider, Messages.EndCheckPermission);
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/resource/aad/plugin.ts:353]: where we send the error telemetry?

File: packages/fx-core/src/plugins/resource/aad/graph.ts (hunk 2 above)
Comment (SLdragon) [reply packages/fx-core/src/plugins/resource/aad/graph.ts:132]: OK

File: packages/fx-core/src/plugins/resource/aad/constants.ts (hunk 1 above)
Comment (SLdragon) [reply packages/fx-core/src/plugins/resource/aad/constants.ts]: Sure

File: packages/fx-core/src/plugins/resource/aad/plugin.ts (hunk 3 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/plugin.ts:353]: Error telemetry will be sent in index.ts. 
For other info that needs to add to telemetry, will update necessary info in telemetry in another pr after we have confirmed required info with yanliang.

**Review by SLdragon — APPROVED** [review]

**Review by adashen — APPROVED** [review]

**Pull request merged**